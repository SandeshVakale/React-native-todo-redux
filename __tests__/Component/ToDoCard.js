import 'react-native';
import React from 'react';
import ToDoCard from '../../src/Components/TodoCard';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

const mockStore = configureStore([]);

test('ToDoCard component renders correctly', () => {
  const initialStore = {
    todoModel: {todo: []},
  };
  let store = mockStore(initialStore);

  let item = {
    name: 'Hello Test',
    id: '12345',
  };

  const tree = renderer
    .create(
      <Provider store={store}>
        <ToDoCard item={item} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
