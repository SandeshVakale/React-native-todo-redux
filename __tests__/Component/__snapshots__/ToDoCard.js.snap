// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoCard component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": 5,
    }
  }
>
  <Text
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "justifyContent": "center",
        "padding": 10,
        "width": "80%",
      }
    }
  >
    Hello Test
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <
      color="rgba(52, 52, 52, 0.8)"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f056",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
          ],
          "iconName": "minus-circle",
          "prefix": "fas",
        }
      }
      size={32}
    />
  </View>
</View>
`;
